torch/distributed/distributed_c10d.py

```diff
@@ -273,32 +273,113 @@ def __getattribute__(self, key):
 reduce_op = _reduce_op()
 
 
-class group(object):
+# DO NOT USE THESE FIELDS DIRECTLY.
+# Use them through the _world object to make sure the _world override mecanism
+_pg_map: Dict[ProcessGroup, Tuple[str, Optional[Store]]] = {}
+_pg_names: Dict[ProcessGroup, str] = {}
+_pg_group_ranks: Dict[ProcessGroup, Dict[int, int]] = {}
+_group_count = 0
```

Comment:
Are these fields only used by `class _World`? Why not put them under that class

Reply:
They are used in some other projects. So we have to keep them. Let me ping you some examples.

Reply:
Hey I am randomly browsed into this piece of code. I am wondering why we need to keep them? I know it's being used in other projects, but I feel since we make all those variables be private variables, shouldn't that mean we are free to change this to whatever we like without worrying about BC?

If it's internal change related, maybe we can just refactor it directly.